src/main/java/uk/nhs/hee/tis/trainee/sync/event/PlacementSpecialtyEventListener.java

```diff
@@ -70,5 +85,48 @@ public void onAfterSave(AfterSaveEvent<PlacementSpecialty> event) {
       }
     }
   }
+
+  /**
+   * Before deleting a PlacementSpecialty, ensure it is cached.
+   *
+   * @param event The before-delete event for the placement specialty.
+   */
+  @Override
+  public void onBeforeDelete(BeforeDeleteEvent<PlacementSpecialty> event) {
+    super.onBeforeDelete(event);
+    String id = event.getSource().getString("_id");
+    PlacementSpecialty placementSpecialty = cache.get(id, PlacementSpecialty.class);
+
+    if (placementSpecialty == null) {
+      Optional<PlacementSpecialty> newPlacementSpecialty = placementSpecialtyService.findById(id);
+      newPlacementSpecialty.ifPresent(psToCache -> cache.put(id, psToCache));
+    }
+  }
+
+  /**
+   * Retrieve the deleted PlacementSpecialty from the cache and sync the updated placement.
+   *
+   * @param event The after-delete event for the placement specialty.
+   */
+  @Override
+  public void onAfterDelete(AfterDeleteEvent<PlacementSpecialty> event) {
+    super.onAfterDelete(event);
+    String id = event.getSource().getString("_id");
```

Comment:
I think ObjectId _ids will throw an error here. This might work better:
String id = event.getSource().get("_id").toString();